public/js/saveApplicant.js

```diff
@@ -32,34 +73,49 @@ let makeApiRequest = async (data) => {
         },
         method: 'post',
         body: JSON.stringify(data)
-    })
-        .then(response => response.json())
-        .then((data) => {
-            if (data.success) {
-                window.location.href = './admin';
-            } else {
-                document.getElementById('errorMsg').innerHTML = "Please contact administrator. Or <a href='./admin'>click here</a> to return to admin page"
-            }
-        })
-}
+    }).then(response => {
+        const generalErrorMessage = document.querySelector('#generalError');
 
-let validateFormInputs = (data) => {
-    let validate = []
+        if (response.status === 200) {
+            window.location.href = './admin';
+        } else if (response.status === 400) {
+            generalErrorMessage.innerHTML = "You must fill out all form options.";
+        } else {
+            generalErrorMessage.innerHTML = "Something went wrong, please try again later.";
+        }
+        generalErrorMessage.classList.remove('hidden');
+        generalErrorMessage.classList.add('alert-danger');
+    });
+};
 
-    validate.push(evalLength(data.name))
-    validate.push(evalLength(data.email))
-    validate.push(evalLength(data.phoneNumber))
-    validate.push(evalLength(data.whyDev))
-    validate.push(evalLength(data.codeExperience))
-    validate.push(evalLength(data.hearAboutId))
+let validateFormInputs = (data) => {
+    let validate = {};
 
-    return validate.includes(false) ? false : true
-}
+    validate.name = {
+        validLengthVarChar: varCharMaxLength(data.name),
+        isName: isName(data.name),
+        isPresent: isPresent(data.name)
+    };
+    validate.email = {
+        validLengthVarChar: varCharMaxLength(data.email),
+        isEmail: isEmail(data.email),
+        isPresent: isPresent(data.email)
+    };
+    validate.phone = {
+        isPhone: isPhoneNumber(data.phoneNumber),
+        isPresent: isPresent(data.phoneNumber)
+    };
+    validate.whyDev = {
+        validLengthText: textAreaMaxLength(data.whyDev),
+        isPresent: isPresent(data.whyDev)
+    };
+    validate.codeExperience = {
+        validLengthText: textAreaMaxLength(data.codeExperience)
+    };
+    validate.notes = {
+        validLengthText: textAreaMaxLength(data.notes)
+    };
```

Comment:
I suppose this could all be like:
```js
valiate = {
    name: {...},
    email: {...},
    ...
}
```

Not sure if this would overwrite things though? Feel free to keep as is.

Reply:
I've changed it to:

    applicantName = {
        validLengthVarChar: varCharMaxLength(data.name),
        isName: isName(data.name),
        isPresent: isPresent(data.name)
    };

    email = {
        validLengthVarChar: varCharMaxLength(data.email),
        isEmail: isEmail(data.email),
        isPresent: isPresent(data.email)
    };

    phone = {
        isPhone: isPhoneNumber(data.phoneNumber),
        isPresent: isPresent(data.phoneNumber)
    };

    whyDev = {
        validLengthText: textAreaMaxLength(data.whyDev),
        isPresent: isPresent(data.whyDev)
    };

    codeExperience = {
        validLengthText: textAreaMaxLength(data.codeExperience)
    };

    notes = {
        validLengthText: textAreaMaxLength(data.notes)
    };

    validate = {
        name: {...applicantName},
        email,
        phone,
        whyDev,
        codeExperience,
        notes
    };
    
Is that what you meant?

Reply:
Sort of, sorry should have been more clear. You can have it all in one:
```js
validate = {
    name: {
        validLengthVarChar: varCharMaxLength(data.name),
        isName: isName(data.name),
        isPresent: isPresent(data.name)
    },
    email: {
        validLengthVarChar: varCharMaxLength(data.email),
        isEmail: isEmail(data.email),
        isPresent: isPresent(data.email)
    },
    phone: {
        isPhone: isPhoneNumber(data.phoneNumber),
        isPresent: isPresent(data.phoneNumber)
    } ,
    whyDev: {
        validLengthText: textAreaMaxLength(data.whyDev),
        isPresent: isPresent(data.whyDev)
    },
    codeExperience: {
        validLengthText: textAreaMaxLength(data.codeExperience)
    },
    notes: {
        validLengthText: textAreaMaxLength(data.notes)
    }
};
```